aten/src/ATen/native/ScatterGatherShapeChecks.h

```diff
@@ -24,17 +24,18 @@ static void gather_shape_check(const Tensor& self, int64_t dim,
   for (int64_t i = 0; i < self_dims; ++i) {
     if (i != dim) {
       TORCH_CHECK(
-        ensure_nonempty_size(index, i) == ensure_nonempty_size(self, i),
+        ensure_nonempty_size(index, i) <= ensure_nonempty_size(self, i),
         "Size does not match at dimension ", i,
-        " get ", ensure_nonempty_size(self, i),
-        " vs ", ensure_nonempty_size(index, i)
+        " expected index ", index.sizes(),
```

Comment:
I don't think you want to change these error messages...
The version accessible from python API only allows `==`. So there would be a mismatch between the error message and the doc here :/

Reply:
OK, makes sense. Especially in the out case, agree.. Maybe I could make a flag in gather_stub to turn this check off for cases like this backward?

Reply:
I think in the `out=` version, the output is resized to the indices size anyway. So even in that case you won't be able to use it :D 

A flag sounds like the right solution yes,